src/workflow-handler.ts

```diff
@@ -78,61 +78,37 @@ export class WorkflowHandler {
 
   async getWorkflowRunStatus(): Promise<WorkflowRunResult> {
     try {
-      const runId = await this.getWorkflowRunId();
-      const response = await this.octokit.rest.actions.getWorkflowRun({
-        owner: this.owner,
-        repo: this.repo,
-        run_id: runId
-      });
-      debug('Workflow Run status', response);
-
-      return {
-        url: response.data.html_url,
-        status: ofStatus(response.data.status),
-        conclusion: ofConclusion(response.data.conclusion)
-      };
+      if (this.runName) {
+        return await this.findWorklowRunFromRunName(this.runName);
+      } else {
+        const runId = await this.getWorkflowRunId();
+        const response = await this.octokit.rest.actions.getWorkflowRun({
+          owner: this.owner,
+          repo: this.repo,
+          run_id: runId
+        });
+        debug('Workflow Run status', response);
+
+        return {
+          url: response.data.html_url,
+          status: ofStatus(response.data.status),
+          conclusion: ofConclusion(response.data.conclusion)
+        };
+      }
 
     } catch (error: any) {
       debug('Workflow Run status error', error);
       throw error;
     }
   }
 
-
-  async getWorkflowRunArtifacts(): Promise<WorkflowRunResult> {
```

Comment:
Please do not remove code that you should not change with this PR.
You PR is to fix issue with runName, not cleaning code. Maybe this code is not used yet but could be later